training/CHAPEL_TUTORIAL.md

```diff
@@ -993,358 +1072,618 @@ Expected Output:
+The chunk of A owned by Locale 1 is: 5 6 7 8
+The chunk of A owned by Locale 3 is: 13 14 15 16
+The chunk of A owned by Locale 2 is: 9 10 11 12
 
-| locale | get_nb | put | put_nb | execute_on | execute_on_nb | cache_get_hits | cache_get_misses | cache_put_misses |
-| -----: | -----: | --: | -----: | ---------: | ------------: | -------------: | ---------------: | ---------------: |
-|      0 |     20 |  20 |     10 |         30 |             3 |             40 |               20 |               10 |
-|      1 |     20 |  20 |     10 |         30 |             0 |             40 |               20 |               10 |
-|      2 |     32 |  20 |     10 |         30 |             0 |             28 |               32 |               10 |
-|      3 |     20 |  20 |     10 |         30 |             0 |             40 |               20 |               10 |
+$ ./tutorial -nl 8
+The chunk of A owned by Locale 0 is: 1 2
+The chunk of A owned by Locale 6 is: 13 14
+The chunk of A owned by Locale 2 is: 5 6
+The chunk of A owned by Locale 7 is: 15 16
+The chunk of A owned by Locale 3 is: 7 8
+The chunk of A owned by Locale 5 is: 11 12
+The chunk of A owned by Locale 4 is: 9 10
+The chunk of A owned by Locale 1 is: 3 4
 ```
-We see the number of `put`s has decreased drastically, and it's way faster!
 
-If we decrease `SIZE`, we see the benefit of the aggregation is outweighed by the cost of setting it up.
-You can play around with different values to see at what point aggregation becomes worthwhile for this example.
+<a id="aggregation"></a>
+## Aggregation
+
+For the best performance we want to minimize how often we move data between locales. But there are situations
+where you need to move lots of data between locales. For example assigning local values into a distributed array
+at random indices. In this case, one locale might be copying multiple elements to every other locale.
+Sending each of those elements one at a time results in a lot of communication. Aggregation is used to mitigate
+this.
+
+I like to think of this in terms of transporting people, where our goal is minimizing the total number of trips.
+If a bunch of people are going to the same place, they could all travel there individually. But we'd have
+fewer total trips if we wait until we have enough people to fill a bus and send a whole group at once.
+
+[Copy aggregation](https://chapel-lang.org/docs/modules/packages/CopyAggregation.html#copyaggregation)
+is the same idea. Instead of copying each value individually, we wait until we have a bunch
+that are all going to the same locale and copy them over all at once.
+
+There are two types of copy aggregation:
+* copying local values into remote variables (`DstAggregator`)
+* copying remote values into local variables (`SrcAggregator`)
+
+It's important to note copy aggregation will only work if at least one side is local.
+Both sides being remote (remote-to-remote aggregation) is not supported.
```

Comment:
I really do think it is helpful to add the description of the different perspectives of how deciding when we are using Source versus Destination Aggregators, based on the relative locale versus remote locale for each case.

Reply:
Yah I think this is a good idea! I could add a bit talking about how to the same action can be seen as either a `GET/SrcAggregator ` or a `PUT/DstAggregator` depending on which locale's perspective we are viewing it from (i.e. which locale is doing the computation

Reply:
updated! I like this much better with extra explanation switching between the locale's perspectives. Thanks @jaketrookman!

Reply:
Love the update.